content/chat/replies.textile

```diff
@@ -0,0 +1,122 @@
+2. Select the app you are working with.
+3. Click *Settings* on the right side of the navigation bar.
+4. Scroll down to *Channel rules* and click *Add new rule*.
+5. Give your new rule a Namespace or provide a specific Channel ID. This should be the primary channel, and not the channel you're using for replies.
+6. Select the *Message interactions enabled* checkbox then click *Create channel rule*.
+
+<a href="/images/realtime/message-interactions.png" target="_blank">
+  <img src="/images/realtime/message-interactions.png" style="max-width:80%" alt="Message interactions in dashboard">
+</a>
+
+Once a channel has message interactions enabled, messages received on that channel will contain a unique @timeSerial@ that can be referenced by later messages. These message replies also add a @ref@ object to message payloads that enables you to indicate what an interaction represents, and which message it is referring to. The @ref@ object contains two fields:
+
+- @type@ (String) := a constant representing the reason for the interaction.
+- @timeSerial@ (String) := a unique identifier used to reference a specific message, assigned when messages are sent in message interaction enabled channels. When replying to a message use the timeSerial of the parent message.
+
+h3(#publish). Publish message replies or threads
+
+With message interactions enabled, each message now automatically has the @timeSerial@ attached, enabling the message to be referenced in replies or threads. To reference a previous message, include the @ref@ object inside the @extras@ object of your payload.
+
+The @extras@ object is a JSON object of arbitrary key-value pairs that may contain metadata such as those required when publishing message replies. When referencing messages the @ref@ object needs to contain the following two key-value pairs:
+
+* A @type@ constant string defining the reason for interacting with the message (replying), for message replies it'll be @chat.reply@, or for message threads it could be @chat.thread@. Note that Ably has reserved strings beginning with @com.ably.@.
+* The @timeserial@ string of the message to interact with.
+
+The following is an example of sending a message reply:
+
+```[javascript]
+function publishReply(message, timeSerial) {
+  channel.publish({
+    name: 'message',
+    data: "This stream is amazing!",
+    extras: {
+      ref: {
+        type: "chat.reply",
+        timeserial: timeSerial
+      }
+    }
+  })
+}
+```
+
+h3(#subscribe). Subscribe to message replies or threads
+
+Subscribe to message replies to receive updates based on channel updates on @refType@. The subscriber for message replies is different from the subscriber for new messages because this subscriber requires a @refType@ to be defined, with the value @chat.reply@. Any event with that @refType@ will be received with this subscriber, enabling you to update the UI where appropriate to display this message reply on the relevant message.
```

Comment:
It's not clear: do we not get replies on the base channel by default, or is this a way to get only the replies? Can we get both replies and new messages on the same subscription?

Reply:
... and if it's the case that we can but don't want to, why?